blog/content/posts/client_registration.md

```diff
@@ -0,0 +1,214 @@
+  - Works with both DCR and CIMD
+
+**Medium attacker cost / Medium mitigation complexity: Localhost impersonation**
+
+- **Attack**: Run malicious app on `localhost:8080`, impersonate legitimate client
+- **Cost**: Trick user into running a malicious application (plus consenting for that app to have data access)
+- **Problem**: Desktop apps can't hold secrets, hard to prove identity
+
+**High attacker cost / High mitigation complexity: OS-attested applications**
+
+- **Attack**: Get malicious app signed by OS vendor
+- **Cost**: Extremely high \- requires compromising OS vendor processes
+- **Mitigation**: OS-level attestation (future work)
+
+### **Solution: Software Statements for Desktop Applications**
+
+For the middle tier, localhost impersonation, we need software statements. This involves:
+
+1. Client hosts a JSON Web Key Set (JWKS) on their backend
+2. Client authenticates the user through their own flow
+3. Backend issues a short-lived, signed JWT attesting to the client's identity
+4. Client includes this JWT in the OAuth flow
+5. Server verifies the JWT against the trusted JWKS
+
+This dramatically raises the bar for impersonation. An attacker would need to:
+
+- Compromise the client's backend infrastructure, or
+- Successfully impersonate the client's authentication flow
+
+Crucially, **software statements work with both DCR and CIMD**. They're not a competing solution \- they're a complementary security layer.
+
+### **Future: OS-Level Attestation**
+
+The strongest protection would be OS-level attestation, e.g. having MacOS, Windows, or Android attest that a piece of software is legitimate. This would make impersonation unreasonably expensive. While the exact way this ties into a software statement is yet to be prototyped, the general shape is evident: thread OS-level identity verification through to the OAuth flow.
+
+## **The Complementary Path Forward**
+
+We're not choosing between solutions. We're exploring complementary approaches for distinct problems:
+
+**For operational issues**: Adding CIMD support in favor of DCR
+
+- Keep DCR for backward compatibility
+- Recommend CIMD for new implementations
+- Both achieve the same authorization goal
+
+**For trust issues**: Layering software statements on top
+
+- Optional enhancement for both DCR and CIMD
+- Required only when localhost impersonation is a concern
+- Servers choose their required trust level
+
+## **Security Considerations**
+
+Both CIMD and software statements require servers to make outbound HTTPS requests, potentially to untrusted domains. Implementations must:
+
+- Prevent SSRF attacks by blocking internal network access
+- Implement timeouts and size limits
+- Consider caching strategies for performance
+- Validate response formats strictly
+
+If we adopt these approaches, we’ll need good best practices and SDK support to help avoid vulnerabilities.
+
+## **Next Steps**
+
+Over the next few weeks, we'll be discussing both approaches in detail:
```

Comment:
Always concerned with articles that are somewhat time bound. I use articles, such as these, as good comparisons between two technologies which may live beyond the short term discussion

Reply:
hmm good point, I can rephrase this line, and remove the "we intend" line below.